src/core/qgsrulebasedlabeling.cpp

```diff
@@ -475,3 +475,44 @@ void QgsRuleBasedLabeling::setSettings( QgsPalLayerSettings *settings, const QSt
       return rule->setSettings( settings );
   }
 }
+
+void QgsRuleBasedLabeling::toSld( QDomNode &parent, const QgsStringMap &props ) const
+{
+  if ( mRootRule == nullptr )
```

Comment:
maybe just `( !mRootRule )`, `nullptr` has the boolean value of false.

Reply:
Oh ok, was not aware, fixed it. 
Just one note, grepping the codebase seems that equality/inequality comparisons against ``nullptr`` are rather common:

````
~/devel/qgis$ git grep -i "== nullptr" | wc -l
44
~/devel/qgis$ git grep -i "!= nullptr" | wc -l
89
````

Reply:
well, not really:
```
ale@littlestar:~/dev/QGIS$ git grep -i "== nullptr" | grep -v astyle| wc -l
6
ale@littlestar:~/dev/QGIS$ git grep -i "!= nullptr" | grep -v astyle| wc -l
29
```
btw, there is not such a big difference, I just find it semantically clearer.